### PR TITLE
support mini fabs

### PR DESCRIPTION
**Pull request by devoncarew**
Add support for mini FABs (40px vs 56px).

### CODE REVIEW

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -1252,7 +1252,7 @@ class RawText extends LeafRenderObjectWidget {
 
 /// A convience widget for paragraphs of text with heterogeneous style.
 ///
-/// The elements paramter is a recusrive list of lists that matches the
+/// The elements paramter is a recursive list of lists that matches the
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: I'm amused that you fixed the typo in "recursive" but not the one in "parameter". :-)

**Comment by Hixie** [comment]
LGTM.

File: packages/flutter/lib/src/widgets/basic.dart (hunk 1 above)
Comment (devoncarew) [inline packages/flutter/lib/src/widgets/basic.dart]: oops, I read it as having the right number of vowels :) will update

File: packages/flutter/lib/src/material/floating_action_button.dart [hunk 2]
```diff
@@ -21,7 +22,8 @@ class FloatingActionButton extends StatefulComponent {
     this.backgroundColor,
     this.elevation: 6,
     this.highlightElevation: 12,
-    this.onPressed
+    this.onPressed,
+    this.mini: false
```
Comment (eseidelGoogle) [inline packages/flutter/lib/src/material/floating_action_button.dart]: Is mini the only size?  Should this be an enum?

Comment (devoncarew) [inline packages/flutter/lib/src/material/floating_action_button.dart]: Yup, just the two, regular and mini. It's just a little ways down on this page: https://www.google.com/design/spec/components/buttons-floating-action-button.html.

**Pull request merged**